universum/modules/structure_handler.py

```diff
@@ -208,6 +212,26 @@ def execute_steps_recursively(self, parent: Optional[Step], cfg: Configuration,
         if child_step_failed:
             raise StepException()
 
+
+    def execute_conditional_step(self, step, step_executor):
+        try:
+            self.execute_one_step(step, step_executor, step.critical)
+            self.close_block()
+            if step.if_succeeded:
+                self.execute_conditional_branch_step(step.if_succeeded, step_executor)
+        except StepException:
+            if step.if_failed:
+                self.close_block()
```

Comment:
Is there alternative to manual opening and closing blocks?
In my opinion it is better to have the same structure of blocks in code and in structure. This is what `run_in_block` function for.

Reply:
If you really want to commit this code, please go ahead. But please also consider immediately refactoring it.